File: day3/puzzle1.py

priorities = 0
lines = open("puzzle.txt", 'r').readlines()

for line in lines:
	if line == '':
		continue
	
	line = line.replace('\n', '')
	
	left  = list( line[: len(line) // 2] )
	right = list( line[len(line) // 2 :] )

	inter = set(left).intersection(right)

	for i in inter:
		code = ord(i)

		if code >= ord('a') and code <= ord('z'):
			priorities += code - ord('a') + 1
		else:
			priorities += code - ord('A') + 27
	pass

print(priorities)

File: day4/puzzle1.js
const fs = require('fs');

function overlap(left, right) {
	const [left1, left2]   = left.split('-');
	const [right1, right2] = right.split('-');
	
	return (
		(parseInt(left1) <= parseInt(right1) && parseInt(left2) >= parseInt(right2)) ||
		(parseInt(left1) >= parseInt(right1) && parseInt(left2) <= parseInt(right2))
	);
}

const lines = fs.readFileSync('puzzle.txt').toString().split('\n');
let counter = 0;

for (line of lines) {
	if (line == '') continue;

	const [left, right] = line.split(',');
	let t = overlap(left, right);

	if (t) counter++;
	console.log(t);
}

console.log(counter);